src/main/java/org/jenkinsci/plugins/mesos/JenkinsScheduler.java

```diff
@@ -556,6 +571,42 @@ private void createMesosTask(Offer offer, Request request) {
 
     results.put(taskId, new Result(request.result, new Mesos.JenkinsSlave(offer.getSlaveId()
         .getValue())));
+    finishedTasks.add(taskId);
+  }
+
+  /**
+   * Checks if the given taskId already exists or just finished running. If it has, then refuse the offer.
+   * @param offer The offer to check if the task is already running
+   * @param taskId The task id
+   * @return True if the task already exists, false otherwise
+   */
+  @VisibleForTesting
+  boolean isExistingTask(Offer offer, TaskID taskId) {
```

Comment:
the name of the function doesn't match its functionality. it's weird that a function that says whether a task exists also refuses the offer. let the function do one thing!